### PR TITLE
Move away from task dependency for plugin-upload on version check sin…

### PR DESCRIPTION
**Pull request by meghameshram**
…ce it executes parallely to plugin-ready

### CODE REVIEW

File: lib/version-check.js [hunk 1]
```diff
@@ -41,7 +41,7 @@ module.exports = function (gulp, gutil) {
 
                 //On success invoke callback function
                 if (typeof cb !== 'undefined') {
-                    cb();
+                    return cb();
```
Comment (dougschroeder) [inline lib/version-check.js]: this is fine for now, but note the guidelines around error handling in node: https://www.joyent.com/developers/node/design/errors

"The usual pattern is that the callback is invoked as callback(err, result), where only one of err and result is non-null, depending on whether the operation succeeded or failed."

File: gulp/plugin.js [hunk 2]
```diff
@@ -101,38 +101,46 @@ module.exports = function (gulp, gutil) {
...
-            pluginUpload = require('../lib/plugin-upload.js')(gulp, gutil);
-          }
-          pluginUpload.upload(server, {
-            debugMode: gutil.env['debug']
-          }).pipe(stream);
-        } else {
-          stream.end();
-        }
-      });
+      return uploadCallBack();
```
Comment (dougschroeder) [inline gulp/plugin.js]: Should you be returning the result of uploadCallback here?  We don't result the result of uploadCallback in the block above.

File: gulp/plugin.js [hunk 3]
```diff
@@ -101,38 +101,46 @@ module.exports = function (gulp, gutil) {
   gulp.task('plugin-upload', ['plugin-ready'], function () {
     var stream = through().obj();
     var server = getPluginServer().getServer();
-    if ((gutil.env['force'] || server.force()) && !gutil.env['prompt']) {
-      if (!pluginUpload) {
-        pluginUpload = require('../lib/plugin-upload.js')(gulp, gutil);
+    var uploadCallBack = function() {
```
Comment (dougschroeder) [inline gulp/plugin.js]: good use of a function here. :)

File: lib/version-check.js (hunk 1 above)
Comment (meghameshram) [inline lib/version-check.js]: Ah ok. but what would be the result object in this case? Just an empty string perhaps?

File: gulp/plugin.js (hunk 2 above)
Comment (meghameshram) [inline gulp/plugin.js]: The earlier code was returning stream. Validate then needs to be refactored to return the result of the uploadCallback too. I will make that change

File: lib/version-check.js (hunk 1 above)
Comment (dougschroeder) [inline lib/version-check.js]: could be "version check passed" or something similar.  Your change above is find for now.

**Pull request merged**